bin/main.ml

```diff
@@ -426,6 +426,36 @@ let runtest =
           $ Arg.(value & pos_all string ["."] name_))
   , Term.info "runtest" ~doc ~man)
 
+let clean =
+  let doc = "Clean the project." in
+  let man =
+    [ `S "DESCRIPTION"
+    ; `P {|Removes files added by jbuilder including _build and .merlin|}
+    ; `Blocks help_secs
+    ]
+  in
+  let go common =
+    set_common common ~targets:[];
+    let log = Log.create () in
+    Future.Scheduler.go ~log
+      begin
+        (* Attach this first before we run anything else *)
+        at_exit (fun () ->
```

Comment:
Ok. In the future there will be a problem with the method used anyway, as computing rules might require running commands. This will be required to support plugins properly.

There is a simpler approach we could use here:
- delete the `_build` directory
- load the trace and delete the files that are in the source tree. The keys in Build_system.Trace.t are the targets generated since the initial build from a clean tree

Then we don't need to create the build system at all. You can expose a function in `Build_system` to read the trace and returns all the targets.